cfgov/v1/migrations/0234_clean_orphaned_revisions.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 3.2.17 on 2023-03-14 13:09
+import logging
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def forwards(apps, schema_editor):
+    Page = apps.get_model('wagtailcore', 'Page')
+    PageRevision = apps.get_model('wagtailcore', 'PageRevision')
+
+    for revision_pk, page_pk in PageRevision.objects.values_list('pk', 'page'):
+        try:
+            page = Page.objects.get(pk=page_pk)
+        except Page.DoesNotExist:
+            page = None
+        else:
+            continue
+
+        revision = PageRevision.objects.get(pk=revision_pk)
+        revision.page = Page(title="Temporary page")
+        logger.info(
+            f"Deleting orphaned page revision with pk {revision_pk} "
+            f"page pk {page_pk}, and path {revision.content['url_path']}."
+        )
+        revision.delete()
```

Comment:
Just a suggestion that might speed up deployment of this change across our various environments (and for local devs), this code does the same thing but takes 6 seconds instead of almost 4 minutes on my laptop:

```suggestion
    orphaned_revisions = PageRevision.objects.exclude(
        page_id__in=Page.objects.values_list("pk", flat=True)
    )

    for pk, page_pk, content in orphaned_revisions.values_list(
        "pk", "page_id", "content"
    ):
        logger.info(
            f"Deleting orphaned page revision with pk {pk} "
            f"page pk {page_pk}, and path {content['url_path']}."
        )

    orphaned_revisions.delete()
```

Calling delete on the queryset instead of instance-by-instance bypasses [the Wagtail code](https://github.com/wagtail/wagtail/blob/fdc36a69891a92db6dca32703cddec1d573c193f/wagtail/models/__init__.py#L2903-L2915) that fails on missing page objects, but I think we don't care about that code getting executed.

Reply:
Ah! That is a very nice optimization!